File: 09 Backspace String Compare/Backspace String Compare.cpp
class Solution
{
public:
    bool backspaceCompare(string S, string T)
    {
        stack<char> s1;
        stack<char> s2;
        for (int i = 0; i < S.size(); i++)
        {
            if (S[i] != '#')
                s1.push(S[i]);
            else
            {
                if (s1.size() != 0)
                    s1.pop();
            }
        }
        for (int i = 0; i < T.size(); i++)
        {
            if (T[i] != '#')
                s2.push(T[i]);
            else
            {
                if (s2.size() != 0)
                    s2.pop();
            }
        }
        int flag = 0;
        if (s1.size() == s2.size())
        {
            while (!s1.empty() && !s2.empty())
            {
                if (s1.top() == s2.top())
                {
                    s1.pop();
                    s2.pop();
                }
                else
                {
                    flag = 1;
                    break;
                }
            }
        }
        else
            return false;
        if (flag == 0)
            return true;
        else
            return false;
    }
};

File: 09 Backspace String Compare/Backspace String Compare.js
/**
 * @param {string} S
 * @param {string} T
 * @return {boolean}
 */
var backspaceCompare = function (S, T) {
    let stack1 = [], stack2 = [];
    for (let i = 0; i < S.length; i++) {
        if (S[i] == '#' && stack1.lenghth != 0)
            stack1.pop();
        else
            stack1.push(S[i]);

    }
    for (let i = 0; i < T.length; i++) {
        if (T[i] == '#' && stack2.lenghth != 0)
            stack2.pop();
        else
            stack2.push(T[i]);

    }
    if (stack1.length != stack2.length)
        return false;
    for (let i = 0; i < stack1.length; i++) {
        if (stack1[i] != stack2[i])
            return false;
    }
    return true;

};